Core/Src/main.c

```diff
@@ -98,6 +142,21 @@ int main(void)
     /* USER CODE END WHILE */
 
     /* USER CODE BEGIN 3 */
+	  while (1)
+	      {
+	        // Initiate a conversion and Read the result of the conversion
+	  		  adc_value = HAL_SPI_TransmitReceive (&hspi1, &command, result, 2, 100); // modify arguments
```

Comment:
This line confuses me, you should be getting the value by passing in a reference to a byte array (which it seems youre doing) not as the return value of the function. the HAL definition of the function is:
> HAL_SPI_TransmitReceive (SPI_HandleTypeDef *hspi, uint8_t *pTxData, uint8_t *pRxData, uint16_t Size, uint32_t Timeout).

iirc it returns the status of the data transfer (which ideally you'd check to make sure the transfer worked). 
You will also probably need to do multiple transfers or read/write multiple bytes I believe.